master/internal/config/authz_config.go

```diff
@@ -0,0 +1,80 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/determined-ai/determined/master/pkg/ptrs"
+)
+
+var (
+	knownAuthZTypes  map[string]bool
+	authZConfigMutex sync.Mutex
+)
+
+// BasicAuthZType is the default authz string id.
+const BasicAuthZType = "basic"
+
+// AuthZConfig is a authz-related section of master config.
+type AuthZConfig struct {
+	Type         string  `json:"type"`
+	FallbackType *string `json:"fallback"`
+}
+
+// DefaultAuthZConfig returns default authz config.
+func DefaultAuthZConfig() *AuthZConfig {
+	return &AuthZConfig{
+		Type: BasicAuthZType,
+		// TODO(ilia): Maybe default to nil?
+		FallbackType: ptrs.Ptr(BasicAuthZType),
+	}
+}
+
+// Validate the authz config.
+func (c *AuthZConfig) Validate() []error {
+	var errs []error
+
+	okTypes := strings.Join(maps.Keys(knownAuthZTypes), ", ")
+	errorTmpl := "\"%s\" is not a known authz type, must be one of: %s"
+	if _, ok := knownAuthZTypes[c.Type]; !ok {
+		errs = append(errs, fmt.Errorf(errorTmpl, c.Type, okTypes))
+	}
+
+	if c.FallbackType != nil {
+		if _, ok := knownAuthZTypes[*c.FallbackType]; !ok {
+			errs = append(errs, fmt.Errorf(errorTmpl, *c.FallbackType, okTypes))
+		}
+	}
+
+	return errs
+}
+
+func initAuthZTypes() {
+	authZConfigMutex.Lock()
+	defer authZConfigMutex.Unlock()
+
+	if knownAuthZTypes != nil {
+		return
+	}
+
+	knownAuthZTypes = make(map[string]bool)
```

Comment:
It seems like `knownAuthZTypes` can be remade if this is called more than once? Should be locked down with a once like in the provider

Reply:
There is a nil check for ```knownAuthZTypes``` above -- which I missed for like a couple minutes wondering how it even worked lol

Reply:
oh lmao I'm just blind